### PR TITLE
Issue 1500 conference validation

### PR DESCRIPTION
**Pull request by gkovats**
For issue #1500 :
The current form has 1 variable to track valid state for the whole form based on formatted_date, so tracking a second field meant tracking state per field. Also just built a `$fields` object to get all the field selectors up front. Didn't know how far this would go.

- conference_url field state now tracked separately, and validated on page load, so meetings with currently bad Zoom urls will either throw an error / highlight the field and disable publish, or automatically correct the url format. 
- moved one formatted_address field error state to this new method, "In person meetings must have a specific address"
- kept language for error messages in PHP so they can use gettext translation.

### Testing:
- load `main` branch, and add a poorly formatted zoom url, then switch to this branch and load the meeting. The Zoom url should get validated immediately - if bad, will error and block publishing. If close, will get corrected and message will say "Url was cleaned up.."
- Updated pointers for Open / Closed checks, so make sure you can't click both types
- Test the validation on Formatted address: 
  - When online meeting, a full address should show the message "Full addresses will trigger meeting is closed message"
  - When in person meeting, an approximate address should throw an error "must have a full address for in person meetings" and block publishing

### CODE REVIEW

File: assets/src/admin-meeting.scss [hunk 1]
```diff
@@ -98,6 +98,17 @@
 		}
 	}
```
Comment (gkovats) [inline assets/src/admin-meeting.scss:100]: used `<small>` tags to follow input fields for immediate error messaging.

File: includes/admin_meeting.php [hunk 2]
```diff
@@ -235,6 +238,12 @@ function () {
             <input value="<?php tsml_echo($location, 'approximate') ?>" type="hidden" name="approximate" id="approximate">
             <input value="<?php tsml_echo($location, 'latitude') ?>" type="hidden" name="latitude" id="latitude">
             <input value="<?php tsml_echo($location, 'latitude') ?>" type="hidden" name="longitude" id="longitude">
+            <small class="error_message" data-message="1">
+                <?php _e('Error: In person meetings must have a specific address.', '12-step-meeting-list') ?>
+            </small>
+            <small class="warning_message" data-message="2">
+                <?php _e('Warning: Unable to process this address for exact location.', '12-step-meeting-list') ?>
+            </small>
```
Comment (gkovats) [inline includes/admin_meeting.php:246]: I added this error state to capture when we get an error from the geocoding call. I figured it was worth displaying, but you may know reasons why this isn't useful - maybe frequent false positives?

File: package.json [hunk 3]
```diff
@@ -1,5 +1,9 @@
 {
 	"private": true,
+    "scripts": {
+        "start": "npx mix watch",
+        "build": "npx mix --production"
+    },
```
Comment (gkovats) [inline package.json:6]: Just added `npm start` as a shortcut for mix watch. I'm lazy.

File: assets/src/admin.js [hunk 4]
```diff
@@ -60,26 +60,86 @@ jQuery(function ($) {
 		//make sure geocoding is finished (basic form validation)
 		var form_valid = true;
```
Comment (gkovats) [inline assets/src/admin.js:62]: Sorry, this is where I went nuts. Made an object to track all the fields, then a small jQuery function to track / attach "state" to each field, and an upper formstate check, so `formatted_address` and `conference_url` (and other fields) could have issues independently.

File: assets/src/admin.js [hunk 5]
```diff
@@ -60,26 +60,86 @@ jQuery(function ($) {
...
+        // after field states update, toggle publish button
+        function updateFormState() {
+            var pendingFields = [];
+            Object.keys($fields).forEach(function(field) {
+                var state = $fields[field].data('state');
+                if (-1 < ['error','loading'].indexOf(state)) {
+                    pendingFields.push($fields[field][0])
+                }
+            });
+            form_valid = !pendingFields.length;
```
Comment (gkovats) [inline assets/src/admin.js:130]: later, we could use `pendingFields` as an array of DOM elements for hyperlinks, to anchor link users to in some top message. "The following fields need attention: ...."

File: assets/src/admin.js [hunk 6]
```diff
@@ -330,17 +388,44 @@ jQuery(function ($) {
...
-			$('input#formatted_address').change();
+		$fields.in_person.on('change', function () {
+			$fields.formatted_address.trigger('change');
 		});
-		$('input#conference_url').change(function () {
-			$('input#formatted_address').change();
-		});
-		$('input#conference_phone').change(function () {
-			$('input#formatted_address').change();
+
```
Comment (gkovats) [inline assets/src/admin.js:394]: Zoom error and cleanup logic

**Review by kiyote33 — APPROVED** [review]
Good job George. Like the radio button effect on the open/close. Nice touch. Everything worked as described in your prep notes. One little thing with the Full Address warning. Shouldn't that be an error instead of warning? You can't really proceed until you fix. Other than that, I think it's good to go.

**Comment by joshreisner** [comment]
i'm really looking forward to reviewing this after NAATW, thanks @gkovats !

**Comment by gkovats** [comment]
Just a heads up, if you're ever waiting on me to merge a PR, let me know - I won't unless asked. Not my codebase, still new.

**Comment by gkovats** [comment]
@kiyote33 Thanks, I missed that - it _was_ an error, but I didn't treat it as such. Just updated in the latest commit.
image

File: assets/src/admin-meeting.scss (hunk 1 above)
Comment (joshreisner) [reply assets/src/admin-meeting.scss:100]: 👍 thanks for using the same color of red we use in other places - maybe eventually we can use css vars for it

File: includes/admin_meeting.php (hunk 2 above)
Comment (joshreisner) [reply includes/admin_meeting.php:246]: no this is great - we can now cross this requirement off our [refactor geocoding](https://github.com/code4recovery/12-step-meeting-list/issues/1002) ticket

File: package.json (hunk 3 above)
Comment (joshreisner) [reply package.json:6]: sounds good - we can make make this standard in other places too

**Review by joshreisner — APPROVED** [review]
works great, and thanks for rebuilding the strings

the only problem i had while testing was my old js and css assets were cached, which meant that i saw the error messages on the screen as black text flush left on the page

i am 99% sure that when this gets released our `TSML_VERSION` bump will bust the cache but it's just something to keep an eye on when this goes out

**Pull request merged**